### ISSUE TITLE
Oscillatory behaviour of fluxes and surface temperature

### ISSUE THREAD

**Issue by nichollsh** [FormingWorlds/PROTEUS#193]
The model sometimes exhibits oscillatory behaviour in `F_atm`, `F_int`, and `T_surf`. This is probably not physical, since it occurs when using the dummy modules. It also occurs when using SPIDER+JANUS.

We should investigate this further. My guess is that it is associated with the time-stepping scheme, which is currently quite crude.

Example from the dummy model in the image below. The behaviour looks particularly bad in this plot because of the log scale on the y-axis. We can see from the OLR lines that the magnitude of the oscillations is quite small (in this case, at least).
![image](https://github.com/user-attachments/assets/599b13ad-177b-48c2-a712-47b1c1908949)

**Comment by timlichtenberg** [comment]
Is this still relevant with the current last version?

**Comment by nichollsh** [comment]
Yeah this is still an issue. It doesn't typically cause problems physically, but we should at least look into using a better time-stepping scheme in terms of model efficiency when the melt fraction is small. Maybe there are techniques that can be borrowed from ESM codes? Somewhat connected to https://github.com/ExPlanetology/aragog/issues/21.

I would suggest we look into this soon, because it could potentially yield significant performance benefits.

**Comment by nichollsh** [comment]
Forcing the model to take large steps shows the following behaviour. This is what we'd expect given the logic above.
<img width="570" alt="image" src="https://github.com/user-attachments/assets/7e4f1340-30ca-437f-833d-f34283fd84ce">
